src/main/java/org/opensearch/knn/index/util/EngineSpecificMethodContext.java

```diff
@@ -25,7 +25,11 @@
  */
 public interface EngineSpecificMethodContext {
 
-    Map<String, Parameter<?>> supportedMethodParameters();
+    Map<String, Parameter<?>> supportedMethodParameters(Context ctx);
 
-    EngineSpecificMethodContext EMPTY = Collections::emptyMap;
+    EngineSpecificMethodContext EMPTY = ctx -> Collections.emptyMap();
+
+    interface Context {
```

Comment:
Lets make this its own pojo and have `QueryContext` We can further reuse this context wherever required. We can use inheritance to further if required down the line, something like `RadialQueryContext` and `HNSWQueryContext`

Reply:
Separated out `QueryContext` from `EngineSpecificMethodContext`. Instead of using `isRadialSearch `, I updated to use `is_ef_search_parameter_supported` to have the check directly. I feel like currently we can use `QueryContext` without inheritance.